SCHEMAS.md

```diff
@@ -0,0 +1,203 @@
+
+This document outlines some potential gotchas when using and consuming the Schemas defined in this library as well as
+some samples below to help with understanding/advanced use cases.
+
+The below links have information about the defining Schemas and how the validation is performed.
+
+[JTD Schemas](https://jsontypedef.com/docs/jtd-in-5-minutes/) for requests and
+responses validation (Via [AJV](https://ajv.js.org/json-type-definition.html)).
+
+## Language
+
+In this document it may use language such as:
+
+- __MUST__ &rarr; this is something you always do
+- __MUST NOT__ &rarr; this is something that you should never do
+- __SHOULD__ &rarr; this is something that is a recommendation and generally is a good idea to follow to avoid issues
+
+## Request/Response types vs single type
+
+You may notice whilst the Schemas are being rolled out that the same object may be used for the request and the response.
+
+However this may not be best practice and in the future it may be better to split out the request and the response
+requests out to have different types.
+
+On the request you would have an optional (i.e. `field?: string | null;`) property on the request.
+
+Then on the response you could include the the field on the response as a required property (i.e. `field: string | null`).
+
+This approach would allow you to introduce a new field with backwards compatibility immediately whilst you update
+consumers to using the new type whilst also emitting the new property in responses.
+
+## Gotchas
+
+### Request/Response type name clash with defined types
+
+When installing the Schemas using the LOKE cli tool if you have request/response objects that are not reference types 
+(i.e. defined using `properties`, `optionalProperties` , etc) then the LOKE cli will create types for you in the
+following format `MethodNameRequest` and `MethodNameResponse`.
+
+When defining your types (in the `definitions` section) you __MUST NOT__ define a type that matches the above format.
+
+For example if you had the following:
+
+ ```typescript
+ // implementation
+interface CreateCustomerRequest {
+  name: string;
+}
+
+interface Customer {
+  id: string;
+  name: string;
+}
+
+CreateCustomer({orgId: string; request: CreateCustomerRequest}): Promise<Customer>
+
+// schema definition
+{
+  ...,
+  definitions: {
+    Customer: {
+      properties: {
+        id: { type: "string" }
+        name: { type: "string" }
+      }
+    },
+    CreateCustomerRequest: {
+      properties: {
+        name: { type: "string" }
+      }
+    }
+  },
+  methods: {
+    CreateCustomer: {
+      requestTypeDef: {
+        properties: {
+          orgId: {type: "string" }
+          request: { ref: "CreateCustomerRequest" }
+        }
+      },
+      responseTypeDef: { ref: "Customer" }
+    }
+  }
+}
+```
+
+When installing the above with the LOKE cli it will generate the following types:
+
+```typescript
+// types created by definitions
+export Type CreateCustomerRequest = {
```

Comment:
There's a few like this. Should this be like below? Don't thing capital `T` works.

```suggestion
export type CreateCustomerRequest = {
```

Reply:
Nice catch without the IDE auto complete just got the casing wrong. Fixed up now